app/controllers/organisations_controller.rb

```diff
@@ -94,10 +94,42 @@ def edit
     end
   end
 
+  def deactivate
+    authorize @organisation
+
+    render "toggle_active", locals: { action: "deactivate" }
+  end
+
+  def reactivate
+    authorize @organisation
+
+    render "toggle_active", locals: { action: "reactivate" }
+  end
+
   def update
-    if current_user.data_coordinator? || current_user.support?
+    if (current_user.data_coordinator? && org_params[:active].nil?) || current_user.support?
       if @organisation.update(org_params)
-        flash[:notice] = I18n.t("organisation.updated")
+        case org_params[:active]
+        when "false"
+          @organisation.users.filter_by_active
```

Comment:
Would it make sense to have something like `@organisation.users.filter_by_active.each(&:deactivate!)` and have a method on the user instead?

Reply:
I think I actually structured it that way originally (since it also allows commonisation with the users controller), then switched to this on the basis that it's more performant. Happy to switch back, though (especially since reactivate is using the less performant approach anyway!)